### DISCUSSION TITLE
Dunno how to make it so when i go to a specific page, an icon changes.

### DISCUSSION THREAD

**Discussion by linomass36** [quasarframework/quasar#12413] [General - Components / Directives / etc] [answered]
I've been able to do this with text using this code:
`<div class="absolute-center">
            <h5>{{ title }}</h5>
            </div>`

`<script>
    export default {
     computed: {
        title() {
        console.log(this.$route)
        let currentPath = this.$route.fullPath
        if (currentPath == '/Settings') return 'Settings'
        else if (currentPath == '/Caclculators') return 'Calculators'
        else if (currentPath == '/Profile') return 'Profile'
        else if (currentPath == '/Exercises') return 'Exercises'
        else if (currentPath == '/Templates') return 'Templates'
        else if (currentPath == '/Achivements') return 'Achivements'
        }
 </script>`

But when trying to do the same with an icon:
`<q-icon name="{{ title2 }}" color="white" size="sm" />`

`<script>
    export default {
     computed: {
        title2() {
        console.log(this.$route)
        let currentPath = this.$route.fullPath
        if (currentPath == '/Settings') return 'settings'
        else if (currentPath == '/Caclculators') return 'calculate'
        else if (currentPath == '/Profile') return 'account_circle'
        else if (currentPath == '/Exercises') return 'fitness_circle'
        else if (currentPath == '/Templates') return 'build'
        else if (currentPath == '/Achivements') return 'emoji_events'
        }
     } 
    }
</script>`

It doesn't work, spitting out in my app an icon that looks sth. like:     }}

I know it's got to do with the quotes after name="", but i dont know how to get past them.

Thanks in advance.

**Comment by metalsadman** [comment, marked as answer]
`:name="title2"`.

**Reply by linomass36** [reply]
Thank you!
Q: what diff does the colon make? why does this work?

**Reply by metalsadman** [reply]
it's a shorthand for `v-bind`, if you don't use a colon it means you are passing your  `title` as a string not the value it holds. https://v3.vuejs.org/api/directives.html#v-bind

**Reply by hawkeye64** [reply]
`v-bind` (or shorthand `:`) tells the Vue SFC compiler to treat what is in the quotes as JavaScript. Otherwise, it's treated as a string.
